package/MDAnalysis/topology/tpr/utils.py

```diff
@@ -423,6 +423,79 @@ def do_mtop(data, fver, tpr_resid_from_one=False):
         elements = Elements(np.array(elements, dtype=object))
         top.add_TopologyAttr(elements)
 
+    # NOTE: the tpr striding code below serves the
+    # purpose of placing us in a suitable "seek" position
+    # in the binary file such that we are well placed
+    # for reading coords and velocities if they are present
+    # the order of operations is based on an analysis of
+    # the C++ code in do_mtop() function in the GROMACS
+    # source at:
+    # src/gromacs/fileio/tpxio.cpp
+    # TODO: expand tpx version support for striding to
+    # the coordinates
+    atnr = ff_params.atnr
+    if fver >= 58:
+        # TODO: the following value is important, and not sure
+        # how to access programmatically yet...
+        # from GMX source code:
+        # api/legacy/include/gromacs/topology/topology_enums.h
+        # worst case scenario we hard code it based on
+        # tpx/GMX version?
```

Comment:
perhaps explicitly mention this PR for more background? Your annotation of the code is really useful and in this way anyone digging into this code can also refer back to them, such as your analysis of data structures etc.